src/Stratis.Bitcoin.Features.BlockStore/BlockStoreCache.cs

```diff
@@ -146,13 +181,23 @@ public void AddToCache(Block block)
             uint256 blockid = block.GetHash();
             this.logger.LogTrace("({0}:'{1}')", nameof(block), blockid);
 
-            Block existingBlock;
-            if (!this.cache.TryGetValue(blockid, out existingBlock))
-                this.cache.Set(blockid, block, TimeSpan.FromMinutes(10));
+            if (!this.cache.TryGetValue(blockid, out Block existingBlock))
+                this.cache.Set(blockid, block, this.blockEntryOptions);
 
             this.logger.LogTrace("(-)[{0}]", existingBlock != null ? "ALREADY_IN_CACHE" : "ADDED_TO_CACHE");
         }
 
+        /// <summary>
+        /// Determine if a block already exists in the cache.
+        /// </summary>
+        /// <param name="blockid">Block id.</param>
+        /// <returns><c>true</c> if the block hash can be found in the cache, otherwise return <c>false</c>.</returns>
+        public bool Exist(uint256 blockid)
```

Comment:
Add these comments to the interface and use `inheritdoc` here.

Reply:
Fixed